WPMedia/fusion-news-theme-blocks
# ASUB-8301 OrderInformation

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

- Update the OfferToProductList component. Every time we [add an item to the cart](https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-2.2.1/blocks/subscriptions-block/components/OfferToProductList/index.jsx#L36):
save the offer campaign,  in session storage ArcXP_campaignName. 
*** Campaigns are unique, if the offer contains more than a single campaign save the one with the oldest end date, the same offer information will be returned when calling GET [/retail/public/v1/offer/live/{campaignName}](https://alc-swagger-template.s3.amazonaws.com/docs/swagger/index.html?url=arc-products/arc-retail.json#/operations/Offers/Get-a-live) regardless of the campaign) for any of these campaigns
***We need a way to retrieve retail info, e.g. price/product description, product/price features, since those will be rendered into the component

Once the item is added. If the user is anonymous then save the sku & priceCode on the session storage ArcXP_cart. Currently, we don't have a way to transfer carts from anonymous user to logged In user. It is causing an anonymous user to be enforced to add the item into the cart again, once he/she is logged In

- Create an OrderInformation component, it will accept as parameters
    a. orderDetail: cart or order detail
    d. offerURL
    e. showOfferURL (boolean)
    f. showPriceDescription (boolean)
   g. ShowProductFeatures (boolean)

**IF** we have the orderDetail info render subtotal, tax & total based on the orderDetail. Otherwise render subtotal, tax & total from getCart();
Price description,  product attributes/features & offerURL can be shown/hidden based on custom fields. 
**IF** the user click on “View subscription offer“ redirect to offerURL



## Jira Ticket

[ASUB-8301]( https://arcpublishing.atlassian.net/browse/ASUB-8301)

## Acceptance Criteria

_copy from ticket_

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout {ASUB-8301_OrderInformation}`
2. Run fusion repo with linked blocks npx fusion start -l @wpmedia/subscriptions-block`
3. <Orderinformation /> component will be rendered inside the checkout component, and we will be using a hook in order to grab the cart or order detail information.
Make sure you have an item in the cart or IF an order was created, you will need to pass the orderNumber if it's the case

```
import React, { useEffect, useState } from "react";

import PropTypes from "@arc-fusion/prop-types";
import OrderInformation from "../../components/OrderInformation";
import useOrder from "../../components/useOrder";

const BLOCK_CLASS_NAME = "b-checkout";

const CheckoutV2 = ({ customFields }) => {
    const { offerURL, successURL, loginURL, stripeIntentsID } = customFields;
    const [orderNumber, setOrderNumber] = useState(); //orderNumber will be obtained on Step 2, once the billing address is included.

    const { cartDetails, orderDetails } = useOrder(orderNumber);

    console.log("cart/order")
    console.log(`orderNumber ${orderNumber}`)
    console.log(cartDetails);
    console.log(orderDetails);

    if (orderNumber && orderDetails?.items?.length) {
        return (
            <OrderInformation
                offerURL={offerURL}
                showOfferURL={false}
                showPriceDescription={true}
                showProductFeatures={false}
                orderDetails={orderDetails}
                className={BLOCK_CLASS_NAME}
            />
        );
    } else if (cartDetails?.items?.length) {
        return (
            <OrderInformation
                offerURL={offerURL}
                showOfferURL={true}
                showPriceDescription={false}
                showProductFeatures={true}
                orderDetails={cartDetails}
                className={BLOCK_CLASS_NAME}
            />
        );
    }

    return null;
};

CheckoutV2.propTypes = {
    customFields: PropTypes.shape({
        offerURL: PropTypes.string.tag({
            defaultValue: "/offer/",
            label: "Offer URL",
        }),
        successURL: PropTypes.string.tag({
            defaultValue: "/",
            label: "Success URL",
        }),
        stripeIntentsID: PropTypes.number.tag({
            label: "Stripe Intents - Provider ID",
            description:
                "In case you have multiple payment providers for stripe Intents, It determines what is the provider ID to be used by default.",
        }),
    }),
};

CheckoutV2.label = "Subscriptions Checkout V2 - Arc Block";
CheckoutV2.icon = "shop-cart";

export default CheckoutV2;
```

## Effect Of Changes

### Before

_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After

_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[ASUB-8301]: https://arcpublishing.atlassian.net/browse/ASUB-8301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
